pkg/apis/management.cattle.io/v3/resource_quota_types.go

```diff
@@ -1,33 +1,97 @@
 package v3
 
+// ProjectResourceQuota represents the allowed and allocated quotas for all namespaces in a project.
 type ProjectResourceQuota struct {
-	Limit     ResourceQuotaLimit `json:"limit,omitempty"`
+	// Limit is the total allowable quota limits shared by all namespaces in the project.
+	// +optional
+	Limit ResourceQuotaLimit `json:"limit,omitempty"`
+
+	// UsedLimit is the currently allocated quota for all namespaces in the project.
+	// +optional
 	UsedLimit ResourceQuotaLimit `json:"usedLimit,omitempty"`
 }
 
+// NamespaceResourceQuota represents the default quota limits for a namespace.
 type NamespaceResourceQuota struct {
+	// Limit is the default quota limits applied to new namespaces.
+	// +optional
 	Limit ResourceQuotaLimit `json:"limit,omitempty"`
 }
 
+// ResourceQuotaLimit holds quota values for different resource.
+// These values are subset of Kubernetes quotas.
+// See https://kubernetes.io/docs/concepts/policy/resource-quotas/ for more details.
 type ResourceQuotaLimit struct {
-	Pods                   string `json:"pods,omitempty"`
-	Services               string `json:"services,omitempty"`
+	// Pods the total number of Pods in a non-terminal state that can exist in the namespace. A pod is in a terminal state if .status.phase in (Failed, Succeeded) is true.
```

Comment:
This is a nit, but above you use the format `<field> is the <explanation>` whereas here you use `<field> the <explanation>`. It would be nice to have a consistent sentence structure. I personally prefer having `is`, but I leave it up to you.